sled-agent/src/instance.rs

```diff
@@ -755,8 +960,18 @@ impl Instance {
             }
         }
 
-        let instance = InstanceInner {
+        let (terminate_tx, terminate_rx) = oneshot::channel();
+        let (tx, rx) = mpsc::channel(QUEUE_SIZE);
+        let (tx_monitor, rx_monitor) = mpsc::channel(1);
```

Comment:
The `InstanceMonitorRunner` waits for the `InstanceRunner` to respond before processing the next state change anyway -- it needs to know how to react -- so I think it's fine that this is only "size of one".